portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,9 +43,13 @@ public void init(){
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    PreparedQuery results = datastore.prepare(new Query("Comment"));
+    int max = getMax(request);
+    if (max < 0) {
+      max = 5; // Default
+    }
+    List<Entity> results = datastore.prepare(new Query("Comment")).asList(FetchOptions.Builder.withLimit(max));
     List<String> messages = new ArrayList<String>();
-    for(Entity entity : results.asIterable()){
+    for(Entity entity : results){
       messages.add((String) entity.getProperty("text"));
```

Comment:
Normally we don't put a space after a cast, so this would be `(String)entity.getProperty("text")`